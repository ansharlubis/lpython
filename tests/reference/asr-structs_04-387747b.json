{
    "basename": "asr-structs_04-387747b",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/structs_04.py",
    "infile_hash": "d1a2040f8609b23e5bde36c06a5f774e02b385911fb601c0d3d1f4bd",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-structs_04-387747b.stdout",
    "stdout_hash": "fd2f8c5edde5d30a8b00c770434140337c556a003e7d8213130354c0",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}