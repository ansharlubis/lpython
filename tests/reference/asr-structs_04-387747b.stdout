([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         A: 
            ([1m[35mDerivedType[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     x: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           x [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     y: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           y [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               A [y x] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m ()), 
         B: 
            ([1m[35mDerivedType[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mPointer[39m[0m 
                              ([1m[35mDerived[39m[0m 1 [33mA[39m [])) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     z: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           z [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               B [a z] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m ()), 
         f: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  4 
                  {
                     b: 
                        ([1m[35mVariable[39m[0m 
                           4 
                           b [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mB[39m []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               f [
               ([1m[35mVar[39m[0m 4 [33mb[39m)] [
               ([1m[35mPrint[39m[0m () [
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33mz[39m 
                     ([1m[35mInteger[39m[0m 4 []) ()) 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33ma[39m 
                        ([1m[35mPointer[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33mx[39m 
                     ([1m[35mInteger[39m[0m 4 []) ()) 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33ma[39m 
                        ([1m[35mPointer[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33my[39m 
                     ([1m[35mReal[39m[0m 4 []) ())] () ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33mz[39m 
                        ([1m[35mInteger[39m[0m 4 []) ()) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mDerivedRef[39m[0m 
                           ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33ma[39m 
                           ([1m[35mPointer[39m[0m 
                              ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33mx[39m 
                        ([1m[35mInteger[39m[0m 4 []) ()) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mRealCompare[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mDerivedRef[39m[0m 
                           ([1m[35mDerivedRef[39m[0m 
                              ([1m[35mVar[39m[0m 4 [33mb[39m) 3 [33ma[39m 
                              ([1m[35mPointer[39m[0m 
                                 ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33my[39m 
                           ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Eq[39m[0m 
                     ([1m[35mRealConstant[39m[0m   3.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.), 
         g: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  5 
                  {
                     a1: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           a1 [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     a2: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           a2 [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mB[39m []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               g [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 5 [33ma1[39m) 
                  ([1m[35mDerivedTypeConstructor[39m[0m 1 [33mA[39m [
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   1.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) 
                        ([1m[35mRealConstant[39m[0m   1.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 4 []))) 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   1.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToInteger[39m[0m 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])))] 
                     ([1m[35mDerived[39m[0m 1 [33mA[39m []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 5 [33ma2[39m) 
                  ([1m[35mDerivedTypeConstructor[39m[0m 1 [33mA[39m [
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   2.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) 
                        ([1m[35mRealConstant[39m[0m   2.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 4 []))) 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mRealConstant[39m[0m   2.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) [1m[32m
                        RealToInteger[39m[0m 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])))] 
                     ([1m[35mDerived[39m[0m 1 [33mA[39m []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 5 [33mb[39m) 
                  ([1m[35mDerivedTypeConstructor[39m[0m 1 [33mB[39m [
                     ([1m[35mVar[39m[0m 5 [33ma1[39m) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))] 
                     ([1m[35mDerived[39m[0m 1 [33mB[39m []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mVar[39m[0m 5 [33mb[39m) 3 [33ma[39m 
                     ([1m[35mPointer[39m[0m 
                        ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 
                  ([1m[35mVar[39m[0m 5 [33ma2[39m) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mVar[39m[0m 5 [33mb[39m) 3 [33mz[39m 
                     ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 5 [33mb[39m) 3 [33ma[39m 
                        ([1m[35mPointer[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33mx[39m 
                     ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mDerivedRef[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 5 [33mb[39m) 3 [33ma[39m 
                        ([1m[35mPointer[39m[0m 
                           ([1m[35mDerived[39m[0m 1 [33mA[39m [])) ()) 2 [33my[39m 
                     ([1m[35mReal[39m[0m 4 []) ()) 
                  ([1m[35mCast[39m[0m 
                     ([1m[35mRealConstant[39m[0m   3.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) [1m[32m
                     RealToReal[39m[0m 
                     ([1m[35mReal[39m[0m 4 []) 
                     ([1m[35mRealConstant[39m[0m   3.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 4 []))) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 5 [33ma1[39m) 2 [33mx[39m 
                        ([1m[35mInteger[39m[0m 4 []) ()) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mRealCompare[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mDerivedRef[39m[0m 
                           ([1m[35mVar[39m[0m 5 [33ma1[39m) 2 [33my[39m 
                           ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Eq[39m[0m 
                     ([1m[35mRealConstant[39m[0m   1.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mDerivedRef[39m[0m 
                        ([1m[35mVar[39m[0m 5 [33ma2[39m) 2 [33mx[39m 
                        ([1m[35mInteger[39m[0m 4 []) ()) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mRealCompare[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mDerivedRef[39m[0m 
                           ([1m[35mVar[39m[0m 5 [33ma2[39m) 2 [33my[39m 
                           ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                        RealToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Eq[39m[0m 
                     ([1m[35mRealConstant[39m[0m   3.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35mSubroutineCall[39m[0m 1 [33mf[39m () [
                  (
                     ([1m[35mVar[39m[0m 5 [33mb[39m))] ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mg[39m () [] ())])
(TranslationUnit (SymbolTable 1 {A: (DerivedType (SymbolTable 2 {x: (Variable 2 x Local () () Default (Integer 4 []) Source Public Required .false.), y: (Variable 2 y Local () () Default (Real 4 []) Source Public Required .false.)}) A [y x] Source Public ()), B: (DerivedType (SymbolTable 3 {a: (Variable 3 a Local () () Default (Pointer (Derived 1 A [])) Source Public Required .false.), z: (Variable 3 z Local () () Default (Integer 4 []) Source Public Required .false.)}) B [a z] Source Public ()), _lpython_main_program: (Subroutine (SymbolTable 7 {}) _lpython_main_program [] [(SubroutineCall 1 g () [] ())] Source Public Implementation () .false. .false.), f: (Subroutine (SymbolTable 4 {b: (Variable 4 b In () () Default (Derived 1 B []) Source Public Required .false.)}) f [(Var 4 b)] [(Print () [(DerivedRef (Var 4 b) 3 z (Integer 4 []) ()) (DerivedRef (DerivedRef (Var 4 b) 3 a (Pointer (Derived 1 A [])) ()) 2 x (Integer 4 []) ()) (DerivedRef (DerivedRef (Var 4 b) 3 a (Pointer (Derived 1 A [])) ()) 2 y (Real 4 []) ())] () ()) (Assert (IntegerCompare (DerivedRef (Var 4 b) 3 z (Integer 4 []) ()) Eq (IntegerConstant 1 (Integer 4 [])) (Logical 4 []) ()) ()) (Assert (IntegerCompare (DerivedRef (DerivedRef (Var 4 b) 3 a (Pointer (Derived 1 A [])) ()) 2 x (Integer 4 []) ()) Eq (IntegerConstant 2 (Integer 4 [])) (Logical 4 []) ()) ()) (Assert (RealCompare (Cast (DerivedRef (DerivedRef (Var 4 b) 3 a (Pointer (Derived 1 A [])) ()) 2 y (Real 4 []) ()) RealToReal (Real 8 []) ()) Eq (RealConstant   3.00000000000000000e+00 (Real 8 [])) (Logical 4 []) ()) ())] Source Public Implementation () .false. .false.), g: (Subroutine (SymbolTable 5 {a1: (Variable 5 a1 Local () () Default (Derived 1 A []) Source Public Required .false.), a2: (Variable 5 a2 Local () () Default (Derived 1 A []) Source Public Required .false.), b: (Variable 5 b Local () () Default (Derived 1 B []) Source Public Required .false.)}) g [] [(= (Var 5 a1) (DerivedTypeConstructor 1 A [(Cast (RealConstant   1.00000000000000000e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   1.00000000000000000e+00 (Real 4 []))) (Cast (RealConstant   1.00000000000000000e+00 (Real 8 [])) RealToInteger (Integer 4 []) (IntegerConstant 1 (Integer 4 [])))] (Derived 1 A []) ()) ()) (= (Var 5 a2) (DerivedTypeConstructor 1 A [(Cast (RealConstant   2.00000000000000000e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   2.00000000000000000e+00 (Real 4 []))) (Cast (RealConstant   2.00000000000000000e+00 (Real 8 [])) RealToInteger (Integer 4 []) (IntegerConstant 2 (Integer 4 [])))] (Derived 1 A []) ()) ()) (= (Var 5 b) (DerivedTypeConstructor 1 B [(Var 5 a1) (IntegerConstant 1 (Integer 4 []))] (Derived 1 B []) ()) ()) (= (DerivedRef (Var 5 b) 3 a (Pointer (Derived 1 A [])) ()) (Var 5 a2) ()) (= (DerivedRef (Var 5 b) 3 z (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 [])) ()) (= (DerivedRef (DerivedRef (Var 5 b) 3 a (Pointer (Derived 1 A [])) ()) 2 x (Integer 4 []) ()) (IntegerConstant 2 (Integer 4 [])) ()) (= (DerivedRef (DerivedRef (Var 5 b) 3 a (Pointer (Derived 1 A [])) ()) 2 y (Real 4 []) ()) (Cast (RealConstant   3.00000000000000000e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   3.00000000000000000e+00 (Real 4 []))) ()) (Assert (IntegerCompare (DerivedRef (Var 5 a1) 2 x (Integer 4 []) ()) Eq (IntegerConstant 1 (Integer 4 [])) (Logical 4 []) ()) ()) (Assert (RealCompare (Cast (DerivedRef (Var 5 a1) 2 y (Real 4 []) ()) RealToReal (Real 8 []) ()) Eq (RealConstant   1.00000000000000000e+00 (Real 8 [])) (Logical 4 []) ()) ()) (Assert (IntegerCompare (DerivedRef (Var 5 a2) 2 x (Integer 4 []) ()) Eq (IntegerConstant 2 (Integer 4 [])) (Logical 4 []) ()) ()) (Assert (RealCompare (Cast (DerivedRef (Var 5 a2) 2 y (Real 4 []) ()) RealToReal (Real 8 []) ()) Eq (RealConstant   3.00000000000000000e+00 (Real 8 [])) (Logical 4 []) ()) ()) (SubroutineCall 1 f () [((Var 5 b))] ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 6 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())])}) [])
